gekitchen/erd/converters/abstract.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic, Optional
+
+T = TypeVar('T')
+
+class ErdValueConverter(Generic[T], ABC):
+    @abstractmethod
+    def erd_encode(self, value: T) -> str:
+        pass
+    @abstractmethod
+    def erd_decode(self, value: str) -> T:
+        pass
```

Comment:
Is there a reason you chose to use an `ABC` over a `Protocol[T]`?  I'm not requesting you change it, just curious.

Reply:
Mainly because I didn't know Protocol[T] existed :).  It's been a long time since I was active in Python... things have moved on from my previous knowledge, still catching back up.